### PR TITLE
add coefficient value to the already existing coefficient column in the CSV export

### PR DESCRIPTION
**Pull request by gravityblast**

### CODE REVIEW

**Review by boudra — APPROVED** [review]

File: src/http/app.ts [hunk 1]
```diff
@@ -153,15 +155,21 @@ app.get("/data/:chainId/rounds/:roundId/vote_coefficients.csv", async (req, res)
...
     const passportScoresMap = processPassportScores(passportScores);
 
     const records = votes.map((vote: any) => {
       const voter = vote.voter.toLowerCase();
-      const combinedVote = { ...vote, ...passportScoresMap[voter] ?? {} };
+      const score = passportScoresMap[voter];
+      const coefficient = score !== undefined ? 1 : 0;
+      const combinedVote = {
+      ...vote,
+      ...(score ?? {}),
```
Comment (boudra) [inline src/http/app.ts:170]: score here is `bool` or `undefined` so it cannot be spread

File: src/http/app.ts [hunk 2]
```diff
@@ -153,15 +155,21 @@ app.get("/data/:chainId/rounds/:roundId/vote_coefficients.csv", async (req, res)
   try {
     const [votes, data] = await Promise.all([
       db.collection(`rounds/${roundId}/votes`).all(),
-      fs.promises.readFile("./data/passport_scores.json", { encoding: "utf8", flag: "r" }),
+      fs.promises.readFile("./data/passport_valid_addresses.json", { encoding: "utf8", flag: "r" }),
```
Comment (boudra) [inline src/http/app.ts:158]: do you think it's useful to return the score as well in the CSV in addition to the coefficient being 1 or 0?

**Review by boudra — CHANGES_REQUESTED** [review]

File: src/http/app.ts (hunk 1 above)
Comment (gravityblast) [reply src/http/app.ts:170]: right I left the same as before, but from the other comment I realized it's better to return all the scores as you said. I'll change it now

File: src/http/app.ts [hunk 3]
```diff
@@ -156,12 +186,18 @@ app.get("/data/:chainId/rounds/:roundId/vote_coefficients.csv", async (req, res)
...
 
     const records = votes.map((vote: any) => {
       const voter = vote.voter.toLowerCase();
-      const combinedVote = { ...vote, ...passportScoresMap[voter] ?? {} };
+      const score = passportScoresMap[voter];
+      const coefficient = score !== undefined ? 1 : 0;
+      const combinedVote = {
+      ...vote,
+      ...(score ?? {}),
+      coefficient,
```
Comment (boudra) [inline src/http/app.ts:199]: score already has a coefficient right? this would be overriding it

Comment (boudra) [reply src/http/app.ts:199]: but in the case that the score is not found in the passport file we want to show 0 as coefficient

Comment (gravityblast) [reply src/http/app.ts:199]: yes I'll push in a second because I changed only the firs part

Comment (gravityblast) [reply src/http/app.ts:199]: I changed it back to draft

**Review by boudra — APPROVED** [review]

File: src/http/app.ts [hunk 4]
```diff
@@ -137,13 +137,43 @@ app.get("/data/:chainId/rounds/:roundId/applications.csv", async (req, res) => {
...
-const processPassportScores = (scores: any[]) => {
+type BasePassportScore = {
+  address: string;
+  score: string | null;
+  status: string;
+  last_score_timestamp: string;
+};
+
+type FullPassportScore = BasePassportScore & {
+  evidence: {
```
[the remark this answers is not in the document]
Comment (gravityblast) [reply src/http/app.ts:148]: @hmrtn good point let me try

**Review by hmrtn — APPROVED** [review]

**Pull request merged**